packages/x-date-pickers/src/DateCalendar/PickersSlideTransition.tsx

```diff
@@ -117,11 +128,13 @@ export function PickersSlideTransition(inProps: SlideTransitionProps) {
     reduceAnimations,
     slideDirection,
     transKey,
-    // extracting `classes` from `other`
-    classes: providedClasses,
+    classes: classesProp,
     ...other
   } = props;
-  const classes = useUtilityClasses(props);
+
+  const { ownerState: pickerOwnerState } = usePickerPrivateContext();
+  const ownerState = { ...pickerOwnerState, slideDirection };
+  const classes = useUtilityClasses(classesProp, ownerState);
```

Comment:
very small performance thing: would it make sense here to prevent creating 2 additional objects?
```suggestion
  const classes = useUtilityClasses(classesProp, {
    ...usePickerPrivateContext().ownerState,
    slideDirection,
  });
```

Reply:
For me we have the exact same amount of objects created with  both approaches.
We only skip the creation of the `ownerState` variable (which is reference based since it's an object so for me it's negligeable)